spec/enum_spec.rb

```diff
@@ -0,0 +1,107 @@
+# This file contains spec tests for Enumerable methods
+
+require '../enum.rb'
+describe Enumerable do
+  let(:empty_array) { [] }
+  let(:array) { [1, 2, 3, 4] }
+  let(:hash) { { 'name' => 'name1' } }
+  let(:range) { (1..4) }
+  describe '#my_each' do
+    it 'return each element of an array' do
+      expect(array.my_each { |element| }).to eql(array.my_each { |element| })
+    end
+    it 'return each element of an hash' do
+      expect(hash.my_each { |element| }).to eql(hash.my_each { |element| })
+    end
+    it 'return each element of a range' do
+      expect(range.my_each { |element| }).to eql(range.my_each { |element| })
```

Comment:
:bulb:  What we are doing here is to check the condition with itself. Instead, we should pass the result into the matcher. The same applies to the my_each_with_index tests.